unity/Holo/Assets/Editor/ModelImport/VTKImport/UnstructuredGridImporter.cs

```diff
@@ -0,0 +1,234 @@
+                {
+                    GetDataflowTangents(streamReader, "alpha");
+                    tangentsAlpha = true;
+                }
+                if (line.IndexOf("LOOKUP_TABLE beta", StringComparison.CurrentCultureIgnoreCase) >= 0)
+                {
+                    GetDataflowTangents(streamReader, "beta");
+                    tangentsBeta = true;
+                }
+                tangentsFlag = tangentsAlpha & tangentsBeta;
+            }
+            if (!verticesFlag | !vectorsFlag | !tangentsFlag)
+            {
+                throw new Exception("Insufficient data in a file!");
+            }
+            SetDataflowIndices();
+        }
+
+        ////Imports a single body mesh file.
+        private void ImportModelBody(StreamReader streamReader)
+        {
+            bool verticesFlag = false;
+            bool indicesFlag = false;
+            while (!streamReader.EndOfStream)
+            {
+                var line = streamReader.ReadLine();
+                if (verticesFlag & indicesFlag)
+                {
+                    break;
+                }
+
+                if (line.IndexOf("POINTS", StringComparison.CurrentCultureIgnoreCase) >= 0)
+                {
+                    string[] pointsData = line.Split(' ');
+                    int numberOfVertices = int.Parse(pointsData[1]);
+                    GetVertices(streamReader, numberOfVertices);
+                    verticesFlag = true;
+                }
+
+                if (line.IndexOf("CELLS", StringComparison.CurrentCultureIgnoreCase) >= 0 & verticesFlag == true)
+                {
+                    string[] cellsData = line.Split(' ');
+                    int numberOfLines = int.Parse(cellsData[1]);
+                    GetBodyIndicesAndNormals(streamReader, numberOfLines);
+                    indicesFlag = true;
+                }
+            }
+        }
+
+        //Get meshes' verices. 
+        private void GetVertices(StreamReader streamReader, int numberOfVertices)
+        {
+            Vertices = new Vector3[numberOfVertices];
+            bool firstVertex = true;
+            for (int i = 0; i < numberOfVertices; i++)
+            {
+                Vector3 currentVertex = streamReader.GetLineVertex();
+                Vertices[i] = currentVertex;
+                BoundingVertices.UpdateBoundingVertices(firstVertex, currentVertex);
+                firstVertex = false;
+            }
+        }
+
+        //Gets body indices in the file and calculates its normals.
+        private void GetBodyIndicesAndNormals(StreamReader streamReader, int numberOfLines)
+        {
+            List<int> indices = new List<int>();
+            List<int> facetIndices = new List<int>();
+
+            bool firstVertex = true;
+            for (int i = 0; i < numberOfLines; i++)
+            {
+                facetIndices = streamReader.GetLineIndices();
+                if (firstVertex)

```

Comment:
`firstVertex` is never changed to `false` so it is pointeless here

Reply:
Fixed.